tests/js/client/http/coordinator-metrics-cluster.js

```diff
@@ -0,0 +1,129 @@
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Valery Mironov
+////////////////////////////////////////////////////////////////////////////////
+
+const jsunity = require("jsunity");
+const arangodb = require("@arangodb");
+const {getEndpointsByType, getMetricRaw} = require("@arangodb/test-helper");
+const parsePrometheusTextFormat = require("parse-prometheus-text-format");
+const _ = require("lodash");
+
+const db = arangodb.db;
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test suite
+////////////////////////////////////////////////////////////////////////////////
+
+function CoordinatorMetricsTestSuite() {
+    return {
+        setUpAll: function () {
+            db._useDatabase("_system");
+            db._create("foo1", {"replication": 3, "numberOfShards": 9});
+            db._create("foo2", {"replication": 2, "numberOfShards": 12});
+            db._create("foo3", {"replication": 1, "numberOfShards": 1});
+            let foo1_values = [];
+            let foo2_values = [{"name": "i"}];
+            let foo3_values = [{"name": "hate"}, {"name": "js"}];
+            for (let i = 0; i !== 100; ++i) {
+                foo1_values.push({"name": i});
+                foo2_values.push({"name": i + 100000});
+                foo3_values.push({"name": i - 100000});
+            }
+            db.foo1.save(foo1_values);
+            db.foo2.save(foo2_values);
+            db.foo3.save(foo3_values);
+            db._createView("foov", "arangosearch", {
+                "links": {
+                    "foo1": {"includeAllFields": true},
+                    "foo2": {"includeAllFields": true},
+                    "foo3": {"includeAllFields": true}
+                }
+            });
+            db._query("FOR d IN foov OPTIONS { waitForSync: true } LIMIT 1 RETURN 1");
+        },
+
+        tearDownAll: function () {
+            db._dropView("foov");
+            db._drop("foo1");
+            db._drop("foo2");
+            db._drop("foo3");
+        },
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief CoordinatorMetricsTestSuite tests
+////////////////////////////////////////////////////////////////////////////////
+
+        testIResearchLinkStats: function () {
+            let coordinators = getEndpointsByType("coordinator");
+            let c = coordinators[0];
+            {// make metrics actual
+                getMetricRaw(c);
+                require('internal').sleep(2);
+            }
```

Comment:
Is this necessary? A few lines below the code calls `getMetricRaw(c)` anyway.

Reply:
May be faster, but yes.
getMetricRaw don't blocking wait answer for cluster metrics, it returns last valid metric and can async call to collect new values